### ISSUE TITLE
CA1305 triggers for methods with overload where `IFormatProvider` parameter is optional

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn-analyzers#6255]
### Analyzer

**Diagnostic ID**: [CA1305](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305): `Specify IFormatProvider`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 7 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 7.0.100](https://dotnet.microsoft.com/download/dotnet/7.0)

### Describe the bug

When a method has a single overload that expects an optional `IFormatProvider` parameter, CA1305 should not be emitted, as one can reasonably assume that the implementation can handle the default value correctly.

### Steps To Reproduce

```csharp
class A
{
    void M()
    {
        C.ConfigureLog(@"A:\log.txt");
    }
}

static class C
{
    public static void ConfigureLog(string fileName, IFormatProvider? formatProvider = null)
    {
    }
}
```

### Expected behavior
No diagnostic.

### Actual behavior
Diagnostic CA1305.

## Additional context

This is a change (regression?) from the .NET 6.0 SDK that broke all of our builds this morning when updating to VS 17.4 (and by extension the 7.0 SDK -- the target framework was still `net6.0` though, so we did not expect breaking changes): the logging framework uses similar calls to configure the log sinks, and each of those calls was marked with CA1305.
Initially it was not clear what the issue was precisely, as the diagnostic spanned the entire method invocation, which is more complex than the sample code above. In addition, the literal `'{0}'` in the diagnostic text did not help narrow down the issue either.

![image](https://user-images.githubusercontent.com/823398/200923068-0474d3e1-5c53-4873-9efd-a5c8d745583b.png)

**Comment by Youssef1313** [comment]
@mavasani Is this by-design?

**Comment by eytan600** [comment]
I have the same problem (vs 17.4.3, sdk 7.0.101), could it be because of the different versions?
Up to what version does it work for sure?
@mavasani, Did you solve it?

**Comment by eytan600** [comment]
I have the same problem (vs 17.4.3, sdk 7.0.101), could it be because of the different versions?
Up to what version does it work for sure?
@jhinder, Did you solve it?

**Comment by eytan600** [comment]
The problem occurs because of SDK version 7.0.
I tried to revert to version 6.0, but it is not possible in Visual Studio.
The problem was solved in two steps that define the use of a previous version of SDK:
**1.** Download previous version of SDK
**2.** Add a configuration file named "global.json" in the folder above the project folder (For example, if project path is D:\USER\Project, the file path wiil be D:\USER\global.json), with the following code: 
{
     "sdk": {
        "version": "number of version"
    }
}

**Comment by hwoodiwiss** [comment]
It looks to me like this Resource string needs changing
https://github.com/dotnet/roslyn-analyzers/blob/c6352bf2e1bd214fce090829de1042000d021497/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx#L1915-L1917

As it doesn't appear to ever be called with a value for `{0}`. https://github.com/search?q=repo%3Adotnet%2Froslyn-analyzers+SpecifyIFormatProviderMessageIFormatProviderOptional&type=code

**Comment by hwoodiwiss** [comment]
Maybe something like:
`The behavior of string formatting could vary based on the current user's locale settings. Provide a value for the optional 'IFormatProvider' argument.`

This would remove the `{0}` red herring.

**Comment by Youssef1313** [comment]
@hwoodiwiss The lack of argument is tracked by https://github.com/dotnet/roslyn-analyzers/issues/6217

**Comment by hwoodiwiss** [comment]
I'll see if I can take a look at that this PM (GMT)

**Comment by cremor** [comment]
This false positive is also triggered by the initial log configuration for Serilog:

```cs
Log.Logger = new LoggerConfiguration()
    .WriteTo.Console() // CA1305 false positive here
    .CreateLogger();
```

One of the (many) optional parameters of the `Console()` method is an `IFormatProvider`.
